src/graph/InterimResult.cpp

```diff
@@ -13,23 +13,27 @@ namespace graph {
 
 constexpr char NotSupported[] = "Type not supported yet";
 
-InterimResult::InterimResult(std::unique_ptr<RowSetWriter> rsWriter) {
-    rsWriter_ = std::move(rsWriter);
-    rsReader_ = std::make_unique<RowSetReader>(rsWriter_->schema(), rsWriter_->data());
-}
-
-
 InterimResult::InterimResult(std::vector<VertexID> vids) {
     vids_ = std::move(vids);
 }
 
+InterimResult::InterimResult(std::vector<std::string> colNames) {
```

Comment:
maybe std::vector<std::string> &&colNames is better

Reply:
Good point.